### PR TITLE
[JFT-02] Create Detail Component

### PR DESCRIPTION
**Pull request by julianfigueroaGL**
This PR will add the detail page for the podcast

Result:
https://share.getcloudapp.com/nOuNYEbZ

### CODE REVIEW

File: src/components/app-detail/app-detail.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <v-card-text class="d-flex flex-no-wrap justify-space-between">
+      <div>
+        <v-card-title class="headline white--text">Detail</v-card-title>
+      </div>
+    </v-card-text>
+    <v-container>
+      <v-row dense>
+        <v-col cols="3">
+          <v-avatar class="ma-3" size="125" tile>
+            <img :src="detailed.artworkUrl600 || detailed.artworkUrl100" />
```
Comment (ogranadagl) [inline src/components/app-detail/app-detail.vue]: Probably you can use a computed property here.

File: src/components/app-detail/app-detail.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          </v-avatar>
+        </v-col>
+        <v-col cols="9">
+          <div>
+            {{ detailed.trackName }}
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: Please use semantic tag

File: src/components/app-detail/app-detail.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <v-spacer />
+            {{ detailed.artistName }}
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: SAB

File: src/components/app-detail/app-detail.vue [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <v-spacer />
+            {{ detailed.collectionName }}
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: SAB

File: src/components/app-detail/app-detail.vue [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            <v-spacer />
+            {{ detailed.country }}
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: SAB

File: src/components/app-detail/app-detail.vue [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            <v-spacer />
+            {{ detailed.primaryGenreName }}
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: SAB

File: src/components/app-detail/app-detail.vue [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+          </div>
+        </v-col>
+      </v-row>
+    </v-container>
+    <v-card-actions> </v-card-actions>
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: Is this WIP?

File: src/views/Detail.vue [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <v-container>
+    <v-row dense>
+      <v-col cols="12">
+        <AppDetail v-bind:detailed="detailed" />
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: Please use shorthand notation https://vuejs.org/v2/guide/syntax.html#v-bind-Shorthand

File: src/views/Detail.vue [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  components: {
+    AppDetail,
+  },
+  data() {
+    return {
+      detailed: {},
+    };
+  },
+  async created() {
+    this.detailed = await api.lookup(Number(this.$route.params.id));
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: Please use parseInt for this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseInt

File: src/views/Detail.vue [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import api from '@/api';
+
+export default {
+  name: 'Detail',
+  components: {
+    AppDetail,
+  },
+  data() {
+    return {
+      detailed: {},
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: I recommend rename **detailed** to **trackDetail**

File: src/views/Detail.vue [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    };
+  },
+  async created() {
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: What about error handling, what happens if I entered an invalid id?

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]
Please update PR title

File: src/components/app-detail/app-detail.vue (hunk 1 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: detailed.artworkUrl600 was removed, data would only contain detailed.artworkUrl100 prop

File: src/components/app-detail/app-detail.vue (hunk 2 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/components/app-detail/app-detail.vue (hunk 3 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/components/app-detail/app-detail.vue (hunk 4 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/components/app-detail/app-detail.vue (hunk 5 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/components/app-detail/app-detail.vue (hunk 6 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/components/app-detail/app-detail.vue (hunk 7 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: The card actions were removed

File: src/views/Detail.vue (hunk 8 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: done

File: src/views/Detail.vue (hunk 9 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: I added an early validation on what's coming from the router params

File: src/views/Detail.vue (hunk 10 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: done

File: src/views/Detail.vue (hunk 11 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: I added an error validation

File: src/components/app-detail/app-detail.vue [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+      <v-container>
+        <v-row dense>
+          <v-col cols="3">
+            <v-avatar class="ma-3" size="125" tile>
+              <img :src="trackDetail.artworkUrl100" />
+            </v-avatar>
+          </v-col>
+          <v-col cols="9">
+            <div>
+              <h2>
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: h2 x 5 times? this doesn't have semantic meaning

File: src/components/app-detail/app-detail.vue [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+                <span>Genre:</span>
+                <span class="font-weight-light"> {{ trackDetail.primaryGenreName }}</span>
+              </h2>
+            </div>
+          </v-col>
+        </v-row>
+      </v-container>
+    </v-card>
+    <v-btn text color="grey" :to="'/'" v-else>
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: I suggest using anchor tag for semantic navigation

File: src/router/index.js [hunk 14]
```diff
@@ -24,4 +30,9 @@ const router = new VueRouter({
   routes,
 });
 
+// eslint-disable-next-line arrow-parens
+router.afterEach(to => {
+  document.title = `GorillaCast - ${to.name}`;
```
Comment (alejandrobenjumeaGL) [inline src/router/index.js]: I recommend using pipe separator `Gorilla Cast | ${to.name}``

File: src/views/Detail.vue [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  async mounted() {
+    this.trackId = Number(this.$route.params.id);
+    let result = {};
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(this.trackId)) {
+      result = await api.lookup(this.trackId);
+    } else {
+      this.snackbar = true;
+    }
+    if (result && result.trackName) {
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: 1. `result` validation is not valid due line 33
2. I suggest changing **trackName** validation to `isEmpty(result)`

File: src/views/Home.vue [hunk 16]
```diff
@@ -23,15 +23,14 @@ export default {
   components: {
     ListPodcast,
   },
-  created() {
-    api.search('software').then((res) => {
-      this.podcasts = res;
-    });
+  async mounted() {
+    const result = await api.search('software');
```
Comment (alejandrobenjumeaGL) [inline src/views/Home.vue]: do we need to define result var?

File: src/views/Detail.vue [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+      trackDetail: {},
+      snackbar: false,
+      trackId: 0,
+    };
+  },
+  async mounted() {
+    this.trackId = Number(this.$route.params.id);
+    let result = {};
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(this.trackId)) {
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: what about refactoring the readability? something like ...

`if (isNaN(this.trackId)) {
      this.snackbar = true;
      return;
    }

    this.trackDetail = await api.lookup(this.trackId);
    
    if (isEmpty(result)) {
      this.snackbar = true;
    }`

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/app-detail/app-detail.vue (hunk 1 above)
Comment (ogranada) [reply src/components/app-detail/app-detail.vue]: 👌

File: src/components/app-detail/app-detail.vue (hunk 12 above)
Comment (ogranada) [reply src/components/app-detail/app-detail.vue]: @julianfigueroaGL Probably you could change the div by a section, in this way the h2 could have a semantic sense.

File: src/components/app-detail/app-detail.vue (hunk 13 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: <router-link> is intended for in-app links (e.g., to a page within your app). Use <a> for external links.

File: src/components/app-detail/app-detail.vue (hunk 12 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/router/index.js (hunk 14 above)
Comment (julianfigueroaGL) [reply src/router/index.js]: done

File: src/views/Detail.vue (hunk 15 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: done

File: src/views/Home.vue (hunk 16 above)
Comment (julianfigueroaGL) [reply src/views/Home.vue]: done

File: src/views/Detail.vue (hunk 17 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: done

File: src/components/app-detail/app-detail.vue [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+          <v-col cols="3">
+            <v-avatar class="ma-3" size="125" tile>
+              <img :src="trackDetail.artworkUrl100" :alt="trackDetail.trackName" />
+            </v-avatar>
+          </v-col>
+          <v-col cols="9">
+            <section>
+              <h2>
+                Name:
+                <span class="font-weight-light">{{ trackDetail.trackName }}</span>
```
Comment (alejandrobenjumeaGL) [inline src/components/app-detail/app-detail.vue]: I suggest using dl
 https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

File: src/views/Detail.vue [hunk 19]
```diff
@@ -0,0 +1,45 @@
+<template>
```
Comment (alejandrobenjumeaGL) [inline src/views/Detail.vue]: Please split files

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/app-detail/app-detail.vue (hunk 18 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue]: done

File: src/views/Detail.vue (hunk 19 above)
Comment (julianfigueroaGL) [reply src/views/Detail.vue]: done

**Review by alejandrobenjumeaGL — APPROVED** [review]

File: src/components/app-detail/app-detail.vue [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+              <dd class="font-weight-light">{{ trackDetail.country }}</dd>
+
+              <dt>Genre</dt>
+              <dd class="font-weight-light">{{ trackDetail.primaryGenreName }}</dd>
+            </dl>
+          </v-col>
+        </v-row>
+      </v-container>
+      <v-card-actions>
+        <v-spacer></v-spacer>
```
Comment (josepenagosGL) [inline src/components/app-detail/app-detail.vue:35]: It'll be nice to use auto close syntax

File: src/components/app-detail/script.js [hunk 21]
```diff
@@ -0,0 +1,14 @@
+import { MESSAGE_PLAY_TRACK, MESSAGE_ADD_FAVORITE } from '@/utils/constants';
+
+export default {
+  name: 'app-detail',
+  props: ['trackDetail'],
```
Comment (josepenagosGL) [inline src/components/app-detail/script.js:5]: As per good practices, it's always recommended to define the type of prop we're expecting

**Review by josepenagosGL — APPROVED** [review]
Man, I left two comments that I'd like you to read just for good practices purposes, but I think it's good so far.

File: src/components/app-detail/app-detail.vue (hunk 20 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/app-detail.vue:35]: yeah, sounds good

File: src/components/app-detail/script.js (hunk 21 above)
Comment (julianfigueroaGL) [reply src/components/app-detail/script.js:5]: got it

**Pull request merged**